File: Scala-TTT/src/main/scala-2.11/com/razorthink/scala/player/Player1.scala
package com.razorthink.scala.player

import com.razorthink.scala.resources.Print


class Player1 (player1Name : String) extends Player{

  val print = new Print;
	playerName = player1Name;
	def getPlayerName : String = return playerName;
	def error : Unit = println("Player 1 Error");
	def getInput:Array[Int] = {
	  var choice = readLine(getPlayerName + "'s chance : ");	  
	  return print.parseInt(choice);
	  
	}
}

File: TicTacToe/src/main/java/com/example/games/ttt/players/Player1.java
package com.example.games.ttt.players;

import com.example.games.ttt.Play;
import com.example.games.ttt.board.Board;

import java.util.Scanner;

/**
 * Created by Prathyush SP on 1/13/16.
 **/
public class Player1 extends Player{

    private static final Scanner scan = new Scanner(System.in);

	public void getName()
	{
		System.out.print("Enter the name of Player 1: ");
		player1 = Play.scan.next();
	}

	public int getInput()
	{		
		System.out.print(Player.player1 + "'s chance: ");
		Board.option = scan.nextLine();
		Board.splits = Board.option.split(" ");
		int x=parseInt(0);
		int y=parseInt(1);				
		if (Board.mat[x][y] == Board.defchar)
			Board.mat[x][y] = 'O';
		else {
			error();
			return -1;
		}	
		return 0;
	}
	
}